src/main/java/com/example/bookstoreapp/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,131 @@
+package com.example.bookstoreapp.service.impl;
+
+import com.example.bookstoreapp.dto.order.OrderDto;
+import com.example.bookstoreapp.dto.order.OrderItemDto;
+import com.example.bookstoreapp.dto.order.OrderRequestDto;
+import com.example.bookstoreapp.dto.order.UpdateOrderStatusRequestDto;
+import com.example.bookstoreapp.exception.EntityNotFoundException;
+import com.example.bookstoreapp.mapper.OrderItemMapper;
+import com.example.bookstoreapp.mapper.OrderMapper;
+import com.example.bookstoreapp.model.CartItem;
+import com.example.bookstoreapp.model.Order;
+import com.example.bookstoreapp.model.OrderItem;
+import com.example.bookstoreapp.model.ShoppingCart;
+import com.example.bookstoreapp.model.User;
+import com.example.bookstoreapp.repository.order.OrderItemRepository;
+import com.example.bookstoreapp.repository.order.OrderRepository;
+import com.example.bookstoreapp.repository.shoppingcart.CartItemRepository;
+import com.example.bookstoreapp.repository.shoppingcart.ShoppingCartRepository;
+import com.example.bookstoreapp.repository.user.UserRepository;
+import com.example.bookstoreapp.service.OrderService;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository cartRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderItemRepository itemRepository;
+    private final OrderMapper orderMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    @Transactional
+    public OrderDto addOrder(OrderRequestDto orderRequestDto) {
+        User user = getUser();
+        ShoppingCart userShoppingCart = findUserShoppingCart();
+        if (userShoppingCart.getCartItems().isEmpty()) {
+            throw new EntityNotFoundException("Cart is empty");
```

Comment:
why u use  EntityNotFoundException if messege equals "Cart is empty". U have to create custom exception for this case

Reply:
agree